ablySpec/RealtimeClient.connection.swift

```diff
@@ -648,6 +648,78 @@ class RealtimeClientConnection: QuickSpec {
                     }
+                }
+
+                // RTN7c
+                context("should deliver as failed the remaining pending messages if") {
+
+                    it("connection is closed") {
+                        client.connect()
+                        defer {
+                            client.dispose()
+                            client.close()
+                        }
+
+                        let channel = client.channel("channel")
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.subscribeToStateChanges { state, status in
+                                if state == .Attached {
+                                    channel.publish("message", cb: { status in
+                                        expect(status.state).to(equal(ARTState.Error))
+                                        expect(status.errorInfo!.code).to(equal(80003))
+                                        done()
+                                    })
+                                    client.close()
+                                }
+                            }
+                            channel.attach()
+                        }
+                    }
+
+                    it("connection state enters FAILED") {
+                        client.connect()
+                        defer {
+                            client.dispose()
+                            client.close()
+                        }
+
+                        let channel = client.channel("channel")
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.subscribeToStateChanges { state, status in
+                                if state == .Attached {
+                                    channel.publish("message", cb: { status in
+                                        expect(status.state).to(equal(ARTState.Error))
+                                        expect(status.errorInfo!.code).to(equal(80000))
+                                        done()
+                                    })
+                                    client.fail()
+                                }
+                            }
+                            channel.attach()
+                        }
+                    }
+
+                    it("lost connection state") {
+                        let client = ARTRealtimeExtended(options: AblyTests.commonAppSetup())
+                        defer { client.close() }
+
+                        let channel = client.channel("channel")
+
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.subscribeToStateChanges { state, status in
+                                if state == .Attached {
+                                    channel.publish("message", cb: { status in
+                                        expect(status.state).to(equal(ARTState.Error))
```

Comment:
I think it's basically the same, but having two arguments that need to be at least typed down explicitly encourages the user to acknowledge the error, so it probably leads to more robust code.

Ideally we would use an enum, so that you need to pattern-match it to extract either the OK value or the error, but there's no such thing in Objective-C, only in Swift.